packages/react-badge/src/components/Badge/Badge.types.ts

```diff
@@ -1,9 +1,9 @@
-import type { IntrinsicShorthandProps, ComponentProps, ComponentState } from '@fluentui/react-utilities';
+import * as React from 'react';
+import type { ComponentProps, ComponentState, ObjectShorthandProps } from '@fluentui/react-utilities';
 
 export type BadgeSlots = {
-  // React internally has a non-standard `color` attribute for HTML elements
-  root: Omit<IntrinsicShorthandProps<'div'>, 'color'>;
-  icon?: IntrinsicShorthandProps<'span'>;
+  root: ObjectShorthandProps<React.HTMLAttributes<HTMLElement>>;
+  icon?: ObjectShorthandProps<React.HTMLAttributes<HTMLElement>>;
```

Comment:
Do this mean that `IntrinsicShorthandProps` type does not work as it should?

Reply:
yes, it's not working as it should, the unions completely break the Avatar when using PresenceBadgeProps

Reply:
Can you please an issue for this problem?

Reply:
I had something like this happen with `SplitButton` yesterday. I can try and pull this branch and do what I did there to see if it works.